components/autowithdraw-shared.js

```diff
@@ -0,0 +1,62 @@
+import { InputGroup } from 'react-bootstrap'
+import { Checkbox, Input } from './form'
+import { useMe } from './me'
+import { useEffect, useState } from 'react'
+import { isNumber } from 'mathjs'
+import { numWithUnits } from '../lib/format'
+
+function autoWithdrawThreshold ({ me }) {
+  return isNumber(me?.privates?.autoWithdrawThreshold) ? me?.privates?.autoWithdrawThreshold : 10000
+}
+
+export function autowithdrawInitial ({ me, priority = false }) {
+  return {
+    priority,
+    autoWithdrawThreshold: autoWithdrawThreshold({ me }),
+    autoWithdrawMaxFeePercent: isNumber(me?.privates?.autoWithdrawMaxFeePercent) ? me?.privates?.autoWithdrawMaxFeePercent : 1
+  }
+}
+
+export function AutowithdrawSettings ({ priority }) {
+  const me = useMe()
+  const threshold = autoWithdrawThreshold({ me })
+
+  const [sendThreshold, setSendThreshold] = useState(Math.max(Math.floor(threshold / 10), 1))
+
+  useEffect(() => {
+    setSendThreshold(Math.max(Math.floor(threshold / 10), 1))
+  }, [autoWithdrawThreshold])
```

Comment:
Did you intended to use `threshold` here as a dependency? Else I don't understand how the reference to `autoWithdrawThreshold` can ever change since it's a pure function outside of this function scope here.

Reply:
probably